Core/src/BareFatJets.cpp

```diff
@@ -54,77 +56,74 @@ void BareFatJets::clear(){
+    t->Branch(jetName + "_subjet","TClonesArray", &subjet, 128000, 0);
+    t->Branch(jetName + "_nSubjets","vector<int>",&nSubjets);
+    t->Branch(jetName + "_firstSubjet","vector<int>",&firstSubjet);
+    t->Branch(jetName + "_subjet_btag","vector<float>",&subjet_btag);
 
-    t->Branch(jetName + "Hbb","vector<float>",&hbb);
+    t->Branch(jetName + "_Hbb","vector<float>",&hbb);
 
-    t->Branch(jetName + "topMVA","vector<float>",&topMVA);
+    t->Branch(jetName + "_topMVA","vector<float>",&topMVA);
 }
 
 void BareFatJets::setBranchAddresses(TTree *t, std::string prefix){
-    TString jetName("fatjet" + prefix);
-    TString algo(prefix);
-    if (prefix == "")
-        algo = "ak8";
 
+    cachedPrefix = prefix;
+    TString jetName("fatjet" + cachedPrefix);
     // init() called by BareP4
     BareP4::setBranchAddresses(t,jetName.Data());
 
-    BareFunctions::SetBranchAddress(t,jetName + "RawPt", &rawPt);
+    BareFunctions::SetBranchAddress(t,jetName + "_RawPt", &rawPt);
 
-    BareFunctions::SetBranchAddress(t,jetName + "Flavour" ,&flavour);
-    BareFunctions::SetBranchAddress(t,jetName + "Tau1"	,&tau1);
-    BareFunctions::SetBranchAddress(t,jetName + "Tau2"	,&tau2);
-    BareFunctions::SetBranchAddress(t,jetName + "Tau3"	,&tau3);
+    BareFunctions::SetBranchAddress(t,jetName + "_Flavour" ,&flavour);
+    BareFunctions::SetBranchAddress(t,jetName + "_Tau1"  ,&tau1);
+    BareFunctions::SetBranchAddress(t,jetName + "_Tau2"  ,&tau2);
+    BareFunctions::SetBranchAddress(t,jetName + "_Tau3"  ,&tau3);
 
-    BareFunctions::SetBranchAddress(t,jetName + "TrimmedMass"	,&trimmedMass);
-    BareFunctions::SetBranchAddress(t,jetName + "PrunedMass"	,&prunedMass);
-    BareFunctions::SetBranchAddress(t,jetName + "FilteredMass"	,&filteredMass);
-    BareFunctions::SetBranchAddress(t,jetName + "SoftdropMass"	,&softdropMass);
+    BareFunctions::SetBranchAddress(t,jetName + "_TrimmedMass"   ,&trimmedMass);
+    BareFunctions::SetBranchAddress(t,jetName + "_PrunedMass"    ,&prunedMass);
+    BareFunctions::SetBranchAddress(t,jetName + "_FilteredMass"  ,&filteredMass);
+    BareFunctions::SetBranchAddress(t,jetName + "_SoftdropMass"  ,&softdropMass);
 
-    BareFunctions::SetBranchAddress(t,algo + "_subjet"	,&ak8_subjet);
-    BareFunctions::SetBranchAddress(t,algo + "jet_hasSubjet",&ak8jet_hasSubjet);
-    BareFunctions::SetBranchAddress(t,algo + "subjet_btag",&ak8subjet_btag);
+    BareFunctions::SetBranchAddress(t,jetName + "_subjet"  ,&subjet);
+    BareFunctions::SetBranchAddress(t,jetName + "_nSubjets",&nSubjets);
+    BareFunctions::SetBranchAddress(t,jetName + "_firstSubjet",&firstSubjet);
+    BareFunctions::SetBranchAddress(t,jetName + "_subjet_btag",&subjet_btag);
 
-    BareFunctions::SetBranchAddress(t,jetName + "Hbb",&hbb);
-    BareFunctions::SetBranchAddress(t,jetName + "topMVA",&topMVA);
+    BareFunctions::SetBranchAddress(t,jetName + "_Hbb",&hbb);
+    BareFunctions::SetBranchAddress(t,jetName + "_topMVA",&topMVA);
 }
```

Comment:
why all the underscores ? I think that other classes like leptons taus photons jets do not have a underscore.